sdks/java/testing/nexmark/src/test/java/org/apache/beam/sdk/nexmark/queries/sql/SqlBoundedSideInputJoinTest.java

```diff
@@ -48,166 +47,182 @@
-        false);
-  }
+    @Test
+    public void queryMatchesModelStreamingDirect() throws Exception {
+      NexmarkConfiguration config = NexmarkConfiguration.DEFAULT.copy();
+      config.sideInputType = NexmarkUtils.SideInputType.DIRECT;
+      config.numEventGenerators = 1;
+      config.numEvents = 5000;
+      config.sideInputRowCount = 10;
+      config.sideInputNumShards = 3;
+      queryMatchesModel(
+          "SqlBoundedSideInputJoinTestStreaming",
+          config,
+          getQuery(config),
+          new BoundedSideInputJoinModel(config),
+          true);
+    }
+
+    @Test
+    public void queryMatchesModelBatchCsv() throws Exception {
+      NexmarkConfiguration config = NexmarkConfiguration.DEFAULT.copy();
+      config.sideInputType = NexmarkUtils.SideInputType.CSV;
+      config.numEventGenerators = 1;
+      config.numEvents = 5000;
+      config.sideInputRowCount = 10;
+      config.sideInputNumShards = 3;
+
+      queryMatchesModel(
+          "SqlBoundedSideInputJoinTestBatch",
+          config,
+          getQuery(config),
+          new BoundedSideInputJoinModel(config),
+          false);
+    }
 
-  @Test
-  @Category(NeedsRunner.class)
-  public void queryMatchesModelStreamingDirect() throws Exception {
-    NexmarkConfiguration config = NexmarkConfiguration.DEFAULT.copy();
-    config.sideInputType = NexmarkUtils.SideInputType.DIRECT;
-    config.numEventGenerators = 1;
-    config.numEvents = 5000;
-    config.sideInputRowCount = 10;
-    config.sideInputNumShards = 3;
-    queryMatchesModel(
-        "SqlBoundedSideInputJoinTestStreaming",
-        config,
-        new SqlBoundedSideInputJoin(config),
-        new BoundedSideInputJoinModel(config),
-        true);
+    @Test
+    public void queryMatchesModelStreamingCsv() throws Exception {
+      NexmarkConfiguration config = NexmarkConfiguration.DEFAULT.copy();
+      config.sideInputType = NexmarkUtils.SideInputType.CSV;
+      config.numEventGenerators = 1;
+      config.numEvents = 5000;
+      config.sideInputRowCount = 10;
+      config.sideInputNumShards = 3;
+      queryMatchesModel(
+          "SqlBoundedSideInputJoinTestStreaming",
+          config,
+          getQuery(config),
+          new BoundedSideInputJoinModel(config),
+          true);
+    }
   }
 
-  @Test
-  @Category(NeedsRunner.class)
-  public void queryMatchesModelBatchCsv() throws Exception {
-    NexmarkConfiguration config = NexmarkConfiguration.DEFAULT.copy();
-    config.sideInputType = NexmarkUtils.SideInputType.CSV;
-    config.numEventGenerators = 1;
-    config.numEvents = 5000;
-    config.sideInputRowCount = 10;
-    config.sideInputNumShards = 3;
-
-    queryMatchesModel(
-        "SqlBoundedSideInputJoinTestBatch",
-        config,
-        new SqlBoundedSideInputJoin(config),
-        new BoundedSideInputJoinModel(config),
-        false);
+  @RunWith(JUnit4.class)
+  public static class SqlBoundedSideInputJoinTest_Calcite extends SqlBoundedSideInputJoinTestCases {
```

Comment:
Checkstyle says no on the underscore. It looks to be failing in a few other places as well.